src/webapp/components/customize-settings-dialog/CustomizeSettingsDialog.tsx

```diff
@@ -0,0 +1,215 @@
+                },
+            },
+        ],
+        [exportTranslation, loading, usecases]
+    );
+
+    const handleTranslationUpload = useCallback(
+        async (_key: string | undefined, lang: string, terms: Record<string, string>) => {
+            await importTranslation(() => usecases.config.importTranslations(lang, terms));
+        },
+        [usecases, importTranslation]
+    );
+
+    return (
+        <ConfirmationDialog
+            isOpen={true}
+            fullWidth={true}
+            onSave={save}
+            cancelText={i18n.t("Close")}
+            onCancel={onClose}
+            disableSave={disableSave}
+        >
+            <ImportTranslationDialog type="custom-text" ref={translationImportRef} onSave={handleTranslationUpload} />
+            <h3>{i18n.t("Home page logo")}</h3>
+            <Row>
+                <IconUpload>
+                    <IconContainer>
+                        <img src={logoVal} alt={`Home page logo`} />
+                    </IconContainer>
+                    <FileInput type="file" onChange={handleFileUpload} />
+                </IconUpload>
+            </Row>
+            <HeaderMenu>
+                <h3>{i18n.t("Customize application text")}</h3>{" "}
+                {!isCustomTextDefault && (
+                    <>
+                        <IconButton onClick={handleClickMenu}>
+                            <MoreVertIcon />
+                        </IconButton>
+                        <Menu anchorEl={menuAnchor} open={menuOpen} onClose={handleCloseMenu}>
+                            {menuAction.map(action => (
+                                <StyledMenuItem key={action.key} onClick={action.onClick}>
+                                    {action.icon}
+                                    {action.text}
+                                </StyledMenuItem>
+                            ))}
+                        </Menu>
+                    </>
+                )}
+            </HeaderMenu>
+
+            {customTextKeys.map(key => (
+                <Row key={key}>
+                    <TextField
+                        fullWidth={true}
+                        label={customTextLabel[key]}
+                        placeholder={i18n.t(defaultCustomText[key].referenceValue)}
+                        value={customTextVal[key] ? customTextVal[key]?.referenceValue : ""}
+                        InputLabelProps={{
+                            shrink: true,
+                        }}
+                        onChange={onChangeField(key)}
+                    />
+                </Row>
+            ))}
+        </ConfirmationDialog>
+    );
+};
+
+const Row = styled.div`
+    margin-bottom: 25px;
+`;
+
+const IconContainer = styled.div`
+    flex-shrink: 0;
+    background-color: #276696;
+    width: 100%;
+    border-radius: 8px;
+    text-align: center;
+
+    img {
+        padding: 10px;
+        max-height: 100px;
+    }
+`;
+
+const IconUpload = styled.div`
```

Comment:
For all this `styled.div` you can use Box and add all styles you want to as attributes. But of course, if you think giving it a name would make it more descriptive is your choice also